tdrs-backend/tdpservice/users/models.py

```diff
@@ -13,58 +15,92 @@
+    login_gov_uuid = models.UUIDField(
+        editable=False, blank=True, null=True, unique=True
+    )
 
     # Unique `hhsid` user claim for AMS OpenID users.
     # In the future, `TokenAuthorizationAMS.get_auth_options` will use `hhs_id` as the primary auth field.
     # See also: CustomAuthentication.py
-    hhs_id = models.CharField(editable=False,
-                              max_length=12,
-                              blank=True,
-                              null=True,
-                              unique=True)
+    hhs_id = models.CharField(
+        editable=False, max_length=12, blank=True, null=True, unique=True
+    )
 
+    # deprecated: use `account_approval_status`
     # Note this is handled differently than `is_active`, which comes from AbstractUser.
     # Django will totally prevent a user with is_active=True from authorizing.
     # This field `deactivated` helps us to notify the user client-side of their status
     # with an "Inactive Account" message.
     deactivated = models.BooleanField(
-        _('deactivated'),
+        _("deactivated"),
         default=False,
         help_text=_(
-            'Designates whether this user should be treated as active. '
-            'Unselect this instead of deleting accounts.'
+            "Deprecated: use Account Approval Status instead - "
+            "Designates whether this user should be treated as active. "
+            "Unselect this instead of deleting accounts."
         ),
     )
 
+    # deprecated: use `account_approval_status` instead
     # This shows access request has been submitted and needs approval. When flag is True, Admin
     # sees the request and has to assign user to group
     access_request = models.BooleanField(
         default=False,
         help_text=_(
-            'Designates whether this user account has requested access to TDP. '
-            'Users with this checked must have groups assigned for the application to work correctly.'
+            "Deprecated: use Account Approval Status instead - "
+            "Designates whether this user account has requested access to TDP. "
+            "Users with this checked must have groups assigned for the application to work correctly."
         ),
     )
 
+    # replaces the deprecated `access_request` and `deactivated` fields above
+    # Designate an account's approval status. options: INITIAL, ACCESS_REQUEST, PENDING, APPROVED, DENIED, DEACTIVATED
+    # property `is_deactivated` below returns True if `account_approval_status` is `DEACTIVATED`
+    account_approval_status = models.CharField(
+        max_length=40,
+        choices=AccountApprovalStatusChoices.choices,
+        default=AccountApprovalStatusChoices.INITIAL,
+        help_text=_(
+            "Designates whether this user account is active and approved to access TDP. "
+            "Users in an approved state are allowed access."
+        ),
+    )
+    _loaded_values = None
+    _adding = True
```

Comment:
I added the above values so they would be initialized. Not being initialized would cause the code to break if  these weren't set.